components/datetime/tests/datetime.rs

```diff
@@ -486,15 +474,15 @@ fn test_dayperiod_patterns() {
                         ]);
                         let dtf = TypedDateTimeFormatter::<Gregorian>::try_new_unstable(
                             &local_provider.as_downcasting(),
-                            &data_locale,
+                            &locale,
                             Default::default(),
                         )
                         .unwrap();
                         assert_writeable_eq!(
                             dtf.format(&datetime),
                             *expected,
                             "\n\
-                            locale:   `{}`,\n\
+                            locale:   `{:?}`,\n\
```

Comment:
Thought/Issue: We probably want to keep a `Display` impl on Preferences?

Reply:
I don't want to make that assumption now. It's not a problem to use `Debug` in a test.